### PR TITLE
[v.0.3] Don't expose 0-dim tensors to Variable API

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
CC @colesbury

File: torch/csrc/autograd/python_variable.cpp [hunk 1]
```diff
@@ -43,6 +43,8 @@ PyObject * THPVariable_Wrap(Variable var)
     Py_RETURN_NONE;
   }
 
+  THPUtils_assert(var.dim() > 0, "variable must have at least 1 dimension");
```
Comment (colesbury) [inline torch/csrc/autograd/python_variable.cpp]: I think throwing an exception would be better here. We don't always check if the return value is null.

File: test/test_autograd.py [hunk 2]
```diff
@@ -995,6 +995,15 @@ def test_setitem_mask(self):
...
         self._test_setitem_tensor((5,), Variable(mask[0]))
 
+    def test_select_sum(self):
+            x = Variable(torch.randn(10), requires_grad=True)
+
+            def func(x):
+                return x.select(0, 1).sum()
+
+            gradcheck(func, [x])
+            gradgradcheck(func, [x])
```
Comment (apaszke) [inline test/test_autograd.py]: Can you add a comment for this test? I don't know what makes a combination of select and sum special

**Comment by gchanan** [comment]
ok, I addressed the review comments.

I also had to add code making sure the grad inputs were not scalars either -- this happens in NN but not the normal autograd functions.

**Comment by gchanan** [comment]
build is trying to merge into master; I'll kick it off again when the build system is fixed.
